LuciBase.py

```diff
@@ -1239,3 +1239,69 @@ def check_luci_path(self):
+        print("#----------------WVT Algorithm----------------#")
+        Pixels = []
+        self.create_snr_map(x_min, x_max, y_min, y_max, method=2, n_threads=1)
+        print("#----------------Algorithm Part 1----------------#")
+        SNR_map = fits.open(self.output_dir+'/'+self.object_name+'_SNR.fits')[0].data
+        SNR_map = SNR_map[x_min:x_max,y_min:y_max]
+        fits.writeto(self.output_dir+'/'+self.object_name+'_SNR.fits', SNR_map, overwrite=True)
+        Pixels, x_min, x_max, y_min, y_max = read_in(self.output_dir+'/'+self.object_name+'_SNR.fits')
+        Nearest_Neighbors(Pixels)
+        Init_bins = Bin_Acc(Pixels, pixel_size, StN_target, roundness_crit)
+        plot_Bins(Init_bins, x_min, x_max, y_min, y_max, StN_target, self.output_dir, "bin_acc")
+        print("#----------------Algorithm Part 2----------------#")
+        Final_Bins = WVT(Init_bins, Pixels, StN_target, ToL, pixel_size, self.output_dir)
+        print("#----------------Algorithm Complete--------------#")
+        plot_Bins(Final_Bins, x_min, x_max, y_min, y_max, StN_target, self.output_dir,"final")
+        Bin_data(Final_Bins, Pixels, x_min, y_min, self.output_dir, "WVT_data")
+        print("#----------------Bin Mapping--------------#")
+        pixel_x = []
+        pixel_y = []
+        bins = []
+        bin_map = np.zeros((x_max-x_min, y_max-y_min))
+        j = 0
+        i = 0
+        with open (self.output_dir+'/WVT_data.txt', 'rt') as myfile:
+            myfile = myfile.readlines()[3:]
+            for myline in myfile:
+                myline = myline.strip(' \n')
+                data = [int(s) for s in myline.split() if s.isdigit()]
+                pixel_x.append(data[0])
+                pixel_y.append(data[1])
+                bins.append(data[2])
+        for pix_x, pix_y in zip(pixel_x,pixel_y):
+            bin_map[pix_x,pix_y] = int(bins[i])
+            i += 1
+        print("#----------------Numpy Bin Mapping--------------#")
+        if not os.path.exists(self.output_dir+'/Numpy_Voronoi_Bins'):
+            os.mkdir(self.output_dir+'/Numpy_Voronoi_Bins')
+        for bin_num in list(range(len(Final_Bins))):
+            print("We're at bin number : ", bin_num)
+            bool_bin_map = bin_map[:].T == bin_num
+            canvas = np.zeros((2048, 2064))
+            canvas[y_min:y_max,x_min:x_max]=bool_bin_map
+            bool_bin_map = canvas.astype(bool)
+            np.save(self.output_dir+'/Numpy_Voronoi_Bins/bool_bin_map_%i'%j, bool_bin_map)
+            bin_axis, bin_sky, bin_fit_dict = self.fit_spectrum_region(lines, fit_function, vel_rel, sigma_rel, region=self.output_dir+'/Numpy_Voronoi_Bins/bool_bin_map_%i.npy'%bin_num, bkg=bkg, bayes_bool=False, uncertainty_bool=False, mean=False)
+            index = np.where(bool_bin_map == True)
+            print(index[0])
+            print(index[1])
+            for a, b in zip(index[0], index[1]):
+                velocities_fits[b,a] = bin_fit_dict['velocities']
```

Comment:
Ok somewhere here is where the error regarding the misalignment of bins is coming from. Why did you flip a and b? Also, for 1 bin, you should manually print out the bin pixels in the mask and verify that that is indeed what index is returning. This will be easier when you have the two functions.